api/src/main/java/com/datatorrent/api/Context.java

```diff
@@ -402,6 +402,10 @@
      */
     Attribute<Boolean> BUFFER_SPOOLING = new Attribute<Boolean>(true);
     /**
+     * Whether to apply back pressure and stall the upstream operator when the downstream operator is slow.
+     */
+    Attribute<Boolean> BUFFER_BACK_PRESSURE = new Attribute<Boolean>(true);
```

Comment:
Per port will not work. There is only a single thread running the operator so if there is back pressure from one port blocking the operator writing to the port, it cannot write to another port.

It can be an operator level setting and I can implement it, but I am not able to see a big benefit over enabling/disabling at the app level as some operators will be able to go ahead but some will wait, overall not achieving any bigger purpose.

I see the advantage in having the ability to enable/disable it at DAG level, because by disabling it we will be able to bring data into the DAG asap which may be needed in some scenarios where input sources cannot hold data for long period of time.